include/net/lwm2m.h

```diff
@@ -94,10 +94,17 @@ struct lwm2m_ctx {
 	 */
 	int tls_tag;
 
+	/** When MBEDTLS SNI is enabled socket must be set with destination
+	 *  hostname server.
+	 */
+	char *desthostname;
+	uint16_t desthostnamelen;
```

Comment:
Do I miss something or are `desthostname` and `desthostnamelen` not set in the parser funcition?

Reply:
now we need set it at application code (like lwm2m_client.c). I don't like this, but i had avoid to allocate a buffer inside peerinfo. Is there a problem to allocate it there?

Reply:
The point is you don't need to allocate - just copy the pointer. The URL is either stored in the Security or Firmware Update object implementation so we just need to copy the location. Current approach won't work if you for instance run bootstrap, and the bootstrap server sets the new URL - you'll still use the BS server hostname if you set this in the application.